libs/ui-lib/lib/cim/components/Agent/BMCForm/BMCForm.tsx

```diff
@@ -0,0 +1,149 @@
+import * as React from 'react';
+import {
+  Alert,
+  AlertActionCloseButton,
+  AlertVariant,
+  Button,
+  ButtonVariant,
+  Form,
+  ModalBoxBody,
+  ModalBoxFooter,
+  TextInputTypes,
+} from '@patternfly/react-core';
+import { Formik, FormikProps, FormikConfig } from 'formik';
+
+import {
+  InputField,
+  getRichTextValidation,
+  RichInputField,
+  BMCValidationMessages,
+} from '../../../../common';
+import { AddBmcValues, BMCFormProps } from '../types';
+import { getErrorMessage } from '../../../../common/utils';
+import { useTranslation } from '../../../../common/hooks/use-translation-wrapper';
+import { getInitValues, getNMState, getValidationSchema } from './validationSchemas';
+import ProvisioningConfigErrorAlert from '../../modals/ProvisioningConfigErrorAlert';
+import { NMStateConfig } from './NMstateConfig';
+
+const BMCForm: React.FC<BMCFormProps> = ({
+  onCreateBMH,
+  onClose,
+  hasDHCP,
+  infraEnv,
+  bmh,
+  nmState,
+  secret,
+  isEdit,
+  usedHostnames,
+  provisioningConfigError,
+}) => {
+  const { t } = useTranslation();
+  const [error, setError] = React.useState<string>();
+
+  const addNmState =
+    infraEnv.metadata?.labels && infraEnv.metadata?.labels['networkType'] === 'static';
+
+  const { initValues, validationSchema } = React.useMemo(() => {
+    const initValues = getInitValues(bmh, nmState, secret, isEdit, addNmState);
+    const validationSchema = getValidationSchema(usedHostnames, initValues.hostname, t);
+    return { initValues, validationSchema };
+  }, [bmh, nmState, secret, isEdit, addNmState, usedHostnames, t]);
```

Comment:
we dont have to memoize values & schema

```suggestion
  const addNmState = infraEnv.metadata?.labels?.networkType === 'static';
  const initValues = getInitValues(bmh, nmState, secret, isEdit, addNmState);
  const validationSchema = getValidationSchema(usedHostnames, initValues.hostname, t);
```